Sources/SwiftKuery/ConnectionPoolConnection.swift

```diff
@@ -52,18 +54,18 @@ public class ConnectionPoolConnection: Connection {
             pool?.release(connection: connection)
         }
     }
-    
+
     // MARK: Connections
     /// Establish a connection with the database.
     ///
     /// - Parameter onCompletion: The function to be called when the connection is established.
-    public func connect(onCompletion: (QueryError?) -> ()) {
-        if let _ = connection {
+    public func connect(onCompletion: @escaping (QueryError?) -> ()) {
+        if let _ = self.connection {
```

Comment:
This function is confusing to me! I _think_ it's trying to say this - am I right??

```swift
if self.connection != nil {
    return onCompletion(nil)
}
guard let connection = self.pool?.take() else {
    return onCompletion(.connection("Failed to get connection"))
}
self.connection = connection
onCompletion(nil)

Reply:
Agreed, updated to enhance readability.